### PR TITLE
[Bug] Fix Memory Leak in Json Load

### LINKED ISSUE #5069: [Bug] BE Memory Leak In Json Load
**Describe the bug**
BE Memory Leak

**To Reproduce**
1. Create a Table
2. Using JSON load multiple times
3. Memory increased from 400M to 3G in 20 mins
4. when i stop json load, mem will stay at 2G
5. I use tcmalloc heap profile，I found a memory leak at StreamLoadPipe::read_one_message

**Screenshots**
Here is the HEAP_PROFILE
58EC867964F7159A66F4526DD67C35D1

I Found the leak Here and run the profile again
image

0192AB9EAA3D761A665A402887057E8F

So I want to use smart pointers to refactor this code

### PR DESCRIPTION
**Pull request by stdpain**
## Proposed changes

fix json load memory leak #5069 

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [x] Bugfix (fix json load memory leak which fixes an [issue-5069)](https://github.com/apache/incubator-doris/issues/5069)
- [x] Code refactor (use unique_ptr instead of native ptr)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have create an issue on (Fix #5069 ), and have described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [x] Any dependent changes have been merged

### CODE REVIEW

File: be/src/exec/local_file_reader.cpp [hunk 1]
```diff
@@ -53,21 +53,17 @@ bool LocalFileReader::closed() {
 }
 
 // Read all bytes
-Status LocalFileReader::read_one_message(uint8_t** buf, size_t* length) {
+Status LocalFileReader::read_one_message(std::unique_ptr<uint8_t[]>& buf, size_t* length) {
```
Comment (imay) [inline be/src/exec/local_file_reader.cpp:56]: Use pointer if the parameter will be changed.
Use reference with const 
```suggestion
Status LocalFileReader::read_one_message(std::unique_ptr<uint8_t[]>* buf, size_t* length) {
```

**Review by imay — CHANGES_REQUESTED** [review]

File: be/src/exec/local_file_reader.cpp (hunk 1 above)
Comment (stdpain) [reply be/src/exec/local_file_reader.cpp]: I will apply it

**Review by HappenLee — DISMISSED** [review]
LGTM

**Review by morningman — DISMISSED** [review]
LGTM

**Review by imay — DISMISSED** [review]

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**